File: frontend/components/notification_card.py
from reactpy import component,html
def NotificationCard(notification):
    priority_color = {
        "Low": "#44ff44",
        "Medium": "#ffaa00",
        "High": "#ff4444",
        "Critical": "#ff0000",
    }.get(notification.get("priority"), "#ffffff")

    return html.div(
        {
            "style": {
                "border": f"2px solid {priority_color}",
                "padding": "15px",
                "borderRadius": "10px",
                "backgroundColor": "#0b0f1e",
                "boxShadow": f"0 0 12px {priority_color}",
                "fontFamily": "Orbitron",
            }},
        html.h3(
            {"style": {"marginBottom": "5px", "color": priority_color}},
            f"⚡ {notification.get('topic')}"
        ),
        html.p(
            {"style": {"marginBottom": "10px", "color": "#ccc"}},
            notification.get("message")
        ),
        html.span({"style": {"fontSize": "12px", "color": "#999"}},
                  f"At: {notification.get('created_at')}"),)

File: legecia-frontend/src/components/NotificationCard.js
// src/components/NotificationCard.js
import { motion } from "framer-motion";

function NotificationCard({ data }) {
  return (
    <motion.div
      initial={{ opacity: 0, y: 15 }}
      animate={{ opacity: 1, y: 0 }}
      style={{
        backdropFilter: "blur(10px)",
        background: "rgba(255,255,255,0.05)",
        border: "1px solid rgba(0,255,255,0.3)",
        borderRadius: "16px",
        padding: "16px",
        boxShadow: "0 0 20px rgba(0,255,255,0.15)",
      }}
    >
      <h5 style={{ color: "#00eaff" }}>
        {data.priority} — {data.topic}
      </h5>
      <p>{data.message}</p>
      <small style={{ opacity: 0.7 }}>
        {new Date(data.created_at).toLocaleString()}
      </small>
    </motion.div>
  );
}

export default NotificationCard;
